multinet/api.py

```diff
@@ -138,72 +138,50 @@ def delete_workspace(workspace: str) -> Any:
 
-    if not node_tables or not edge_table:
-        body = request.data.decode("utf8")
-        raise MalformedRequestBody(body)
-
-    missing = [
-        arg[0]
-        for arg in [("node_tables", node_tables), ("edge_table", edge_table)]
-        if arg[1] is None
-    ]
-    if missing:
-        raise RequiredParamsMissing(missing)
+    if not edge_table:
+        raise RequiredParamsMissing(["edge_table"])
 
     loaded_workspace = db.db(workspace)
     if loaded_workspace.has_graph(graph):
         raise AlreadyExists("Graph", graph)
 
-    existing_tables = set([x["name"] for x in loaded_workspace.collections()])
-    edges = loaded_workspace.collection(edge_table).all()
+    # Get reference tables with respective referenced keys,
+    # tables in the _from column and tables in the _to column
+    edge_table_properties = util.get_edge_table_properties(workspace, edge_table)
+    referenced_tables = edge_table_properties["table_keys"]
+    from_tables = edge_table_properties["from_tables"]
+    to_tables = edge_table_properties["to_tables"]
 
-    # Iterate through each edge and check for undefined tables
     errors = []
-    valid_tables: Dict[str, Set[str]] = dict()
-    invalid_tables = set()
-    for edge in edges:
-        nodes = (edge["_from"].split("/"), edge["_to"].split("/"))
-
-        for (table, key) in nodes:
-            if table not in existing_tables:
-                invalid_tables.add(table)
-            elif table in valid_tables:
-                valid_tables[table].add(key)
-            else:
-                valid_tables[table] = {key}
-
-    if invalid_tables:
-        for table in invalid_tables:
+    for table, keys in referenced_tables.items():
+        if not loaded_workspace.has_collection(table):
             errors.append(f"Reference to undefined table: {table}")
+        else:
+            table_keys = set(loaded_workspace.collection(table).keys())
+            undefined = keys - table_keys
 
-    # Iterate through each node table and check for nonexistent keys
-    for table in valid_tables:
-        existing_keys = set(
-            [x["_key"] for x in loaded_workspace.collection(table).all()]
-        )
-        nonexistent_keys = valid_tables[table] - existing_keys
-
-        if len(nonexistent_keys) > 0:
-            errors.append(
-                f"Nonexistent keys {', '.join(nonexistent_keys)} "
-                f"referenced in table: {table}"
-            )
+            if undefined:
+                errors.append(
+                    f"Nonexistent keys {', '.join(undefined)} "
+                    f"referenced in table: {table}"
+                )
 
     # TODO: Update this with the proper JSON schema
     if errors:
         raise ValidationFailed(errors)
 
-    db.create_graph(workspace, graph, node_tables, edge_table)
+    db.create_graph(
+        workspace,
+        graph,
+        edge_table,
+        from_vertex_collections=from_tables,
+        to_vertex_collections=to_tables,
```

Comment:
If you change the argument declarations in the `create_graph` function, then you can pass these two arguments directly as positional arguments, rather than having to use the kwarg style.

Reply:
Alright. I kept them that way for readability, but I'm not really concerned about it.

Reply:
I'm definitely more concerned about `arango` not being the last argument in the signature, and when that's fixed I'm not even sure we'd be able to pass these args as keywords. But even then, I think it actually degrades readability to only pass some of the arguments as keywords without a good reason.